test/tftp/tftp_test.rb

```diff
@@ -49,4 +49,16 @@ def test_boot_filename_has_no_dash_when_prefix_ends_with_slash
   def test_boot_filename_uses_dash_when_prefix_does_not_end_with_slash
     assert_equal "a/b/c-somefile", Proxy::TFTP.boot_filename('a/b/c', '/d/somefile')
   end
+
+  def test_choose_protocol_and_fetch_wget
+    ::Proxy::HttpDownload.any_instance.expects(:start).returns(true).times(3)
+    %w(http://proxy.test https://proxy.test ftp://proxy.test).each do |src|
+      Proxy::TFTP.choose_protocol_and_fetch src, '/destination'
+    end
+  end
+
+  def test_choose_protocol_and_fetch_nfs
+    Proxy::TFTP.logger.expects(:debug).once
```

Comment:
I think the fact that we are logging at a specific level isn't important/is fragile here. I think a check that no exception is raised is in this case clearer in its intention and also less fragile.  Additionally, could add a test that verifies that an exception is raised when the url scheme isn't recognized?